src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -68,19 +108,23 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
         };
     }
 
-    @computed public get onSelectDataSet() {
+    @bind
+    public onSelectProfileX(option:any) {
         if (this.props.store.molecularProfileIdToMolecularProfile.isComplete) {
-            return (option:any)=>{
-                this._selectedMolecularProfile = this.props.store.molecularProfileIdToMolecularProfile.result[option.value];
-            }
-        } else {
-            return (option:any)=>{};
+            this._selectedProfileX = option.value;
         }
     }
 
     @bind
-    private onSelectGene(entrezGeneId:string) {
-        this._selectedEntrezGeneId = parseInt(entrezGeneId, 10);
+    public onSelectProfileY(option:any) {
+        if (this.props.store.molecularProfileIdToMolecularProfile.isComplete) {
```

Comment:
theres no need to wait for this promise to be ready - you're not using it's result on the inside of the if statement